### PR TITLE
Feature/add validate traj exec param

### PR DESCRIPTION
**Pull request by srsidd**
### Description
This addresses #840.

### Checklist
- [x] **Required**: Code is auto formatted using [clang-format](http://moveit.ros.org/documentation/contributing/code)
- [ ] Extended the tutorials / documentation, if necessary [reference](http://moveit.ros.org/documentation/contributing/)
- [ ] Include a screenshot if changing a GUI
- [ ] Optional: Created tests, which fail without this PR [reference](http://docs.ros.org/kinetic/api/moveit_tutorials/html/doc/tests.html)
- [ ] Optional: Decide if this should be cherry-picked to other current ROS branches (Indigo, Jade, Kinetic)

[//]: # "You can expect a response from a maintainer within 7 days. If you haven't heard anything by then, feel free to ping the thread. Thank you!"

### CODE REVIEW

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 1]
```diff
@@ -935,8 +941,11 @@ bool TrajectoryExecutionManager::distributeTrajectory(const moveit_msgs::RobotTr
 
 bool TrajectoryExecutionManager::validate(const TrajectoryExecutionContext& context) const
 {
-  if (allowed_start_tolerance_ == 0)  // skip validation on this magic number
+  // skip validation
+  if (not validate_trajectory_complete_) {
+    ROS_DEBUG_NAMED(name_, "Not validating if trajectory execution is complete");
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: This validation concerns deviations of start start from current state. It's not related to completion!
This change should be reverted!

File: moveit_ros/planning/trajectory_execution_manager/include/moveit/trajectory_execution_manager/trajectory_execution_manager.h [hunk 2]
```diff
@@ -348,6 +351,7 @@ class TrajectoryExecutionManager
   double allowed_goal_duration_margin_;
   double allowed_start_tolerance_;  // joint tolerance for validate(): radians for revolute joints
   double execution_velocity_scaling_;
+  bool validate_trajectory_complete_;
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/include/moveit/trajectory_execution_manager/trajectory_execution_manager.h]: In #840 we discussed to have a double parameter, e.g. `allowed_goal_tolerance_`, similar to `allowed_start_tolerance_`. They should be initialized from each other if only one is specified.
However, as this parameter is dynamically reconfigurable, that's probably not feasible: they are always explicitly set by `DynamicReconfigure`.
If we keep a bool here, I suggest `wait_for_completion_`

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 3]
```diff
@@ -71,6 +71,7 @@ class TrajectoryExecutionManager::DynamicReconfigureImpl
     owner_->setAllowedGoalDurationMargin(config.allowed_goal_duration_margin);
     owner_->setExecutionVelocityScaling(config.execution_velocity_scaling);
     owner_->setAllowedStartTolerance(config.allowed_start_tolerance);
+    owner_->enableValidateTrajectoryExecution(config.validate_trajectory_complete);
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: Use name `setWaitForCompletion()`?

File: moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp [hunk 4]
```diff
@@ -1511,8 +1520,11 @@ bool TrajectoryExecutionManager::executePart(std::size_t part_index)
 
 bool TrajectoryExecutionManager::waitForRobotToStop(const TrajectoryExecutionContext& context, double wait_time)
 {
-  if (allowed_start_tolerance_ == 0)  // skip validation on this magic number
+  // skip validation
```
Comment (rhaschke) [inline moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: Here the check should read
`if (allowed_start_tolerance_ == 0 || !wait_for_completion_)`

Comment (srsidd) [reply moveit_ros/planning/trajectory_execution_manager/src/trajectory_execution_manager.cpp]: Fixed.

**Pull request merged**